### PR TITLE
Fix incorrect background pixmap rendering

### PR DESCRIPTION
**Pull request by marvin0102**
Fix the incorrect rendering of PNG files and pass the test of PNG color formats include RGB, RGBA, PALETTE and GRAY.

### CODE REVIEW

**Review by jserv — CHANGES_REQUESTED** [review]
Run `clang-format -i src/*.[ch]` and squash the git commits.

**Review by jserv — CHANGES_REQUESTED** [review]
You should fetch the latest `main` branch and run `git rebase -i`. See https://verdantfox.com/blog/how-to-git-rebase-mainmaster-onto-your-feature-branch-even-with-merge-conflicts

File: src/image-png.c [hunk 1]
```diff
@@ -2,6 +2,7 @@
 #include <png.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <stdio.h>
```
Comment (jserv) [inline src/image-png.c:5]: Avoid unnecessary header inclusion.

**Review by jserv — CHANGES_REQUESTED** [review]
Read https://cbea.ms/git-commit/ carefully and refine your git commit messages accordingly.

**Comment by jserv** [comment]
@marvin0102, You are still failing to use `git rebase` properly.

Here are the steps you can follow:
```shell
$ git remote add upstream https://github.com/sysprog21/mado
$ git fetch upstream
$ git rebase upstream/main 
```

Then, check the order of commits by running `git log -n 2`
```
commit 59a1a07e866fcffff89b9fdfb27ed14f3349676a (HEAD -> main)
Author: marvin <marvin890102@gmail.com>
Date:   Thu Jul 18 10:55:30 2024 +0800

    Fix the incorrect rendering of PNG file
    
    The RGB color formats of libpng and TWIN are inconsistent.
    The RGB format in libpng is shown as BGR format in TWIN. Add the
    transform function '_convertBGRtoARGB' to correct it.

commit b0a8bf0ff9c4307a1a8930d0f8c6236c571b33e1 (upstream/main)
Author: Jim Huang <jserv.tw@gmail.com>
Date:   Thu Jul 18 14:27:19 2024 +0800

    Simplify the build procedure
    
    Only object files should be tracked in advance, and all C source files
    can be detected via the build targets.
```

Your commit is now appearing after commit b0a8bf0ff9c4307a1a8930d0f8c6236c571b33e1, which is the latest HEAD of `main` branch.

**Comment by jserv** [comment]
Tested on Ubuntu Linux 20.04:

comparison

TWIN attempted to render the scaled `assets/tux.png` (left), while the same PNG file is rendered by the ImageMagick tool (right). The color was incorrect though.

**Comment by marvin0102** [comment]
linux seems not to have the problem of RGB invert. I don't have a linux device nearby. Can you please test for me?

File: src/image-png.c [hunk 2]
```diff
@@ -17,6 +17,25 @@ static void _twin_png_read(png_structp png, png_bytep data, png_size_t size)
         png_error(png, "end of file !\n");
 }
 
+/* FIXME: Utilize pixman or similar accelerated routine to convert */
+#if __APPLE__
```
Comment (jserv) [inline src/image-png.c:21]: Replace with `#if defined(__APPLE__)`.

**Comment by jserv** [comment]
> linux seems not to have the problem of RGB invert. I don't have a linux device nearby. Can you please test for me?

The background rendering works on Ubuntu Linux now.
twin-sdl-linux

**Comment by jserv** [comment]
> linux seems not to have the problem of RGB invert. I don't have a linux device nearby.

Can you try various PNG files as well? Once we verify, the proposed changes can be checked in.

**Comment by marvin0102** [comment]
I already test PNG files with different color formats include RGBA, PALETTE, GRAY. The screenshots are the results of mado and the original PNG files rendered by mac Preview.
截圖 2024-07-19 上午11 19 17
截圖 2024-07-19 上午11 20 09
截圖 2024-07-19 上午11 21 57
截圖 2024-07-19 上午11 22 50

**Comment by jserv** [comment]
Thank @marvin0102 for contributing!

**Pull request merged**